packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,217 @@
+          details: {
+            coding: [
+              {
+                system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                code: "MISSING_VALUE",
+                display: "Missing required fields"
+              }
+            ]
+          }
+        }
+      ]
+    }
+    return {
+      statusCode: 400,
+      body: JSON.stringify(errorResponseBody),
+      headers: {
+        "Content-Type": "application/fhir+json",
+        "Cache-Control": "no-cache"
+      }
+    }
+  }
+
+  const responseBundle: any = {
+    resourceType: "Bundle",
+    id: uuidv4(),
+    meta: {
+      lastUpdated: new Date().toISOString()
+    },
+    type: "transaction-response",
+    entry: []
+  }
+
+  for (const entry of entries) {
+    logger.info("Processing entry", {entry: entry})
+    const entry_resource = entry.resource
+
+    const dynamoDBItem: DynamoDBItem = {
+      RequestID: xRequestId,
+      PrescriptionID: entry_resource.basedOn?.[0]?.identifier?.value,
+      PatientNHSNumber: entry_resource.for?.identifier?.value,
+      PharmacyODSCode: entry_resource.owner?.identifier?.value,
+      TaskID: entry_resource.id,
+      LineItemID: entry_resource.focus?.identifier?.value,
+      TerminalStatusIndicator: entry_resource.status,
+      LastModified: entry_resource.lastModified,
+      Timestamp: new Date().toISOString(),
+      RequestMessage: entry_resource
+    }
+
+    const invalidFields = []
+    for (const [field, value] of Object.entries(dynamoDBItem)) {
+      if (!value) {
+        logger.info("Invalid value", {field: field, value: value})
+        invalidFields.push(field)
+      }
+    }
+
+    if (invalidFields.length > 0) {
+      const errorMessage = `Missing required fields: ${invalidFields.join(", ")}`
+      logger.error("Error message", {errorMessage: errorMessage})
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                  code: "MISSING_VALUE",
+                  display: errorMessage
+                }
+              ]
+            }
+          }
+        ]
+      }
+      return {
+        statusCode: 400,
+        body: JSON.stringify(errorResponseBody),
+        headers: {
+          "Content-Type": "application/fhir+json",
+          "Cache-Control": "no-cache"
+        }
+      }
+    }
+
+    const item = marshall(dynamoDBItem)
+    logger.info("Marshalled item", {item: item})
+
+    const command = new PutItemCommand({
+      TableName: tableName,
+      Item: item
+    })
+    logger.info("Sending PutItemCommand", {command: command})
+    await client.send(command)
+
+    const taskResponse = {
+      response: {
+        status: "201 Created",
+        location: `Task/${dynamoDBItem.TaskID}/_history/1`,
```

Comment:
This location isn't real, the field is optional so we should just remove it

Reply:
The `location` field has been removed.